tests/api-tests/queries/validation.ts

```diff
@@ -1,30 +1,50 @@
-const { visit, Kind } = require('graphql');
+import { visit, Kind, ValidationContext, ASTNode } from 'graphql';
 
-const definitionLimit = maxDefinitions => validationContext => {
+type Fragment = { name: string; atDepth: number };
+
+type Definition = {
+  node: ASTNode;
+  fragments: Fragment[];
+  fieldDepth: number;
+  numFields: number;
+  totalDepth?: number | null;
+  totalNumFields?: number | null;
+};
+
+export const definitionLimit = (maxDefinitions: number) => (
+  validationContext: ValidationContext
+) => {
   const doc = validationContext.getDocument();
   if (doc.definitions.length > maxDefinitions) {
     validationContext.reportError(
-      new Error(`Request contains ${doc.definitions.length} definitions (max: ${maxDefinitions})`),
-      doc
+      // @ts-ignore
```

Comment:
`validationContext.reportError` claims that it wants a `GraphQLError`, but when we give it one we end up with internal server errors. I couldn't work out why without going way deeper than I have time for, so for now we'll stick with what works.